router-tests/testenv/testenv.go

```diff
@@ -423,6 +436,21 @@ func createTestEnv(t testing.TB, cfg *Config) (*Environment, error) {
 		return nil, err
 	}
 
+	if !cfg.ForcePubSubChecks {
+		for _, datasource := range routerConfig.EngineConfig.DatasourceConfigurations {
+			if customEvents := datasource.CustomEvents; customEvents != nil {
+				customEvents.DontVerify = true
+			}
+		}
+		for _, ffConfig := range routerConfig.FeatureFlagConfigs.GetConfigByFeatureFlagName() {
+			for _, datasource := range ffConfig.EngineConfig.DatasourceConfigurations {
+				if customEvents := datasource.CustomEvents; customEvents != nil {
+					customEvents.DontVerify = true
```

Comment:
I'm not sure why we need this flag. It is also not the right way to extend the node_pb definition when this information is set at the router.

Reply:
+1 why do we need to disable this validation?

The only case is when you don't have edfs configuration at all, but then buildPubSub configuration won't be called

If we really need such an option I would suggest to name it differently - smth like

```
DisableEventProviderIdExistenceValidation
```

Just something more describing an intent of that option

Reply:
Create a separate graph configuration without the EDFS graph. In that case, you don't have to deal with cross-concern issues.